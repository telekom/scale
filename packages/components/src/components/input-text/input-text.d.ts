export declare class InputText {
    private getCssClassMap;
    render(): any;
}
