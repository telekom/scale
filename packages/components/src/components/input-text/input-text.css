/**
 * @prop --input-text-border: Border of the input-text
 * @prop --input-text-border-radius: Border radius of the input-text 
 * @prop --input-text-padding: Padding of the input-text
 * @prop --input-text-font-size: Font size of the input-text
 * @prop --input-text-font-family: Font family of the input-text
 * @prop --input-text-validation-font-size: Font size of the validation of the input-text
 * @prop --input-text-validation-opacity: Opacity of the validation of the input-text
 * @prop --input-text-validation-margin: Margin of the validation of the input-text
 * @prop --input-text-validation-color: Color of the validation text of the input-text
 */

.input-text {
  display: flex;
  flex-direction: column;
}

.input-text__input {
  border: var(--input-text-border, 1px solid #dfdfdf);
  border-radius: var(--input-text-border-radius, 4px);
  padding: var(--input-text-padding, 0.5rem);
  font-size: var(--input-text-font-size, 16px);
  font-family: var(--input-text-font-family, inherit);
}

/*
 * TODO: implement accessability
 * -----------------------------
 * Once the user starts tabbing it probably
 * means that a screen-reader is required
 */

.input-text:not(.tabbing):focus {
  outline: 0;
}

.input-text__validation {
  font-size: var(--input-text-validation-font-size, 12px);
  opacity: var(--input-text-validation-opacity, 0.7);
  margin: var(--input-text-validation-margin, 0.5rem 0 0 0);
  color: var(--input-text-validation-color, red);
}
