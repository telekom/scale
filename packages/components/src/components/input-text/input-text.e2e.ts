import { newE2EPage } from '@stencil/core/testing';

describe('t-input-text', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-input-text/>');
    const element = await page.find('t-input-text');
    expect(element).toHaveClass('hydrated');
  });
});
