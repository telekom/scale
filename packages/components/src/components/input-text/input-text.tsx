import { Component, Prop, Event, h, EventEmitter, State } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import {
  Validator,
  getValidator,
  defaultValidator,
  ValidatorEntry,
} from '../../validators';

@Component({
  tag: 't-input-text',
  styleUrl: 'input-text.css',
  shadow: true,
})
export class InputText {
  /** (optional) Input text class */
  @Prop() public customClass?: string = '';
  /** (optional) Input text theme */
  @Prop() public theme?: string = '';
  /** (optional) Input text value */
  @Prop({ mutable: true }) public value: string;
  /** (optional) Input text validator */
  @Prop() public validator: Array<string | ValidatorEntry | Validator<string>>;
  /** (optional) Input text event changed */
  @Event() public changed: EventEmitter<string>;
  /** (optional) Input text state touched */
  @State() public touched: boolean = false;

  // tslint:disable-next-line: variable-name
  public _validator: Validator<string> = defaultValidator;

  public componentWillLoad() {
    this._validator = getValidator<string>(this.validator);
  }

  public componentWillUpdate() {
    this._validator = getValidator<string>(this.validator);
  }

  public handleChange(event) {
    this.value = event.target ? event.target.value : null;
    this.changed.emit(this.value);
    this.touched = true;
  }

  public render() {
    return (
      <div class={this.getCssClassMap()}>
        <input
          type="text"
          class="input-text__input"
          value={this.value}
          onInput={event => this.handleChange(event)}
        />
        {this.touched && !this._validator.validate(this.value) ? (
          <span class="input-text__validation">
            {this._validator.errorMessage}
          </span>
        ) : null}
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'input-text',
      this.customClass && this.customClass,
      this.theme && `input-text--theme-${this.theme}`
    );
  }
}
