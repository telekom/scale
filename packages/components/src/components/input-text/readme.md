# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute | Description | Type                                                | Default     |
| ----------- | --------- | ----------- | --------------------------------------------------- | ----------- |
| `theme`     | `theme`   |             | `string`                                            | `undefined` |
| `validator` | --        |             | `(string \| ValidatorEntry \| Validator<string>)[]` | `undefined` |
| `value`     | `value`   |             | `string`                                            | `undefined` |


## Events

| Event     | Description | Type                  |
| --------- | ----------- | --------------------- |
| `changed` |             | `CustomEvent<string>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
