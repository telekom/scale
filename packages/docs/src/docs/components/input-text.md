---
title: Input Text
description: Input Text element
url: /docs/input-text
contributors:
  - SylviaGonschior
---

# Input Text

Input Text component

selector: `t-input-text`

## Default

```javascript
<t-input-text></t-input-text>
```

<div class="demo-container">
  <t-input-text></t-input-text>
</div>

## With required validator

```javascript
<t-input-text id="input-required-validator"></t-input-text>
```

<div class="demo-container">
  <t-input-text id="input-required-validator"></t-input-text>
  <script>
        const inputRequiredValidator = document.getElementById(
            'input-required-validator'
        );
        inputRequiredValidator.validator = ['required'];
  </script>
</div>

## With number validator

```javascript
<t-input-text id="input-number-validator"></t-input-text>
```

<div class="demo-container">
  <t-input-text id="input-number-validator"></t-input-text>
  <script>
        const inputNumberValidator = document.getElementById(
            'input-number-validator'
        );
        inputNumberValidator.validator = ['number'];
  </script>
</div>

## With Email validator

```javascript
<t-input-text id="input-email-validator"></t-input-text>
```

<div class="demo-container">
  <t-input-text id="input-email-validator"></t-input-text>
  <script>
        const inputEmailValidator = document.getElementById(
            'input-email-validator'
        );
        inputEmailValidator.validator = ['email'];
  </script>
</div>

## With multiple validator

```javascript
<t-input-text id="input-multiple-validators"></t-input-text>
```

<div class="demo-container">
  <t-input-text id="input-multiple-validators"></t-input-text>
  <script>
        const inputMultipleValidators = document.getElementById(
            'input-multiple-validators'
        );
        inputMultipleValidators.validator = [
            'required',
            'email', {
                name: 'length',
                options: {
                    min: 6,
                },
            },
        ];
    </script>
</div>
