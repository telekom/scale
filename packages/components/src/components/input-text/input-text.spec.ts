import { newSpecPage } from '@stencil/core/testing';
import { InputText } from './input-text';

describe('InputText', () => {
  let element;
  beforeEach(async () => {
    element = new InputText();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputText],
      html: `<t-input-text>Label</t-input-text>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('input-text--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toBe('input-text');
  });
});
