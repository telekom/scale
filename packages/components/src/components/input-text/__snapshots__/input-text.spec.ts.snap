// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should match snapshot 1`] = `
<t-input-text>
  <mock:shadow-root>
    <div>
      <div class="input-text">
        <input class="input-text__input" type="text">
      </div>
    </div>
  </mock:shadow-root>
  Label
</t-input-text>
`;
